core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -176,7 +192,17 @@ public void validateIndexName(String index, ClusterState state) {
         }
     }
 
-    public void createIndex(final CreateIndexClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void createIndex(final CreateIndexClusterStateUpdateRequest request,
+                            final ActionListener<ClusterStateUpdateResponse> listener) {
+
+        final int numShardCopies = IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.get(request.settings()) + 1;
+        if (request.waitForActiveShards().exceeds(numShardCopies)) {
```

Comment:
I think it makes sense that the Rollover API will wait. The Shrink API should also wait IMO, but sometimes it may copy lots of files (if hard links are not supported by the FS).. I'm inclined to say both should wait and if we see that the Shrink API times out more often than not, we can revisit.  I didn't read the PR here to voice a proper opinion whether it's simpler to have these 3 actions wait independently (using the same share "waiting" class/cluster state predicate) or have it uniform here (with the risk of people waiting where the shouldn't). My feeling says wait in the action - but that's for your consideration.